### ISSUE TITLE
Allow other User roles access to Admin Panel

### ISSUE THREAD

**Issue by piere129** [strapi/strapi#1152]
**What is the expected behavior?**

As of strapi 12.1.3, the administrator role is the only role that is allowed access to the Admin UI. I think it would be beneficial if we could define in the user role permissions whether or not they can access the Admin UI, and possibly restrict some access on this Admin panel. 

- [x] I'm sure that this issue hasn't already been referenced.

**Comment by Aurelsicoko** [comment]
I'm closing the issue because the feature is one of our top priorities after the stabilization part and it's the most up-voted feature on the community vote page.

**Comment by g-div** [comment]
Any news about this feature ? Can we follow the development progress somewhere (maybe this issue shouldn't be closed) ?

**Comment by g-div** [comment]
In #931 @soupette proposed a work-a-round documented [here](https://github.com/strapi/strapi/blob/master/packages/strapi-admin/doc/customize-menu.md), but it seems to work only on *Content Types* and not on *plugins* and *general* section, am I right or are there other possible solutions already available ?

**Comment by Aurelsicoko** [comment]
@g-div The specifications are almost ready on our side. However, we need to improve the plugin architecture & flexibility to make it work properly. We're currently developing the Documentation plugin, the next one will certainly be the Administrators & Roles plugin.

We can't wait to release this feature as well...

**Comment by g-div** [comment]
Just asking because it was unclear if the feature was already released or not, since all related issues have been already closed. I was otherwise curious if it was possible to follow its development in detail, in order to eventually offer some help there.

**Comment by carlosmarinho** [comment]
Any news if this feature was released? ...

**Comment by derrickmehaffy** [comment]
@carlosmarinho nope not yet, and probably not in the next few months.

**Comment by Aurelsicoko** [comment]
Please see https://portal.productboard.com/strapi/c/8-administrators-roles-permissions for more details.

**Comment by dtaalbers** [comment]
Any idea when this is implemented? It's quite critical for me.

**Comment by mnlbox** [comment]
Based on https://portal.productboard.com/strapi/tabs/3-planned it's planned for Q3 2019 :disappointed:

**Comment by dtaalbers** [comment]
😮 A highly voted feature like this? Ah well, we might change from Strapi in that case. Thanks for the reply!

**Comment by derrickmehaffy** [comment]
@dtaalbers There is *a lot* more that has to happen before the adminUI permissions rework comes. The Strapi team are working as fast as they can. 

Trust me they are aware how popular it is, but they are focusing on the plugin framework rewrite to allow them and others to build great plugins like that.

**Comment by dtaalbers** [comment]
@derrickmehaffy No worries, I understand. Wasn't aware of the rework, I only was a bit surprised about the roadmap seeing the votes for the feature. 👍

**Comment by Aurelsicoko** [comment]
Yes, it's the most asked feature on ProductBoard. However, we have to rework the plugin framework first, test the core and update the architecture, then we'll be able to add new features 👍

**Comment by FrancisVarga** [comment]
is there any update on this?

**Comment by derrickmehaffy** [comment]
@FrancisVarga "Soon"

**Comment by LucasMatuszewski** [comment]
@derrickmehaffy "Soon" === 1 month? 3 months? ;)

**Comment by derrickmehaffy** [comment]
@LucasMatuszewski I won't work for Strapi, so I have no ETA for you. I do know the feature is important to them but their focus is entirely on the plugin framework rewrite at the moment.

**Comment by LucasMatuszewski** [comment]
Thanks for replying so quickly. On Roadmap I see 4Q19.

**Comment by Aurelsicoko** [comment]
FYI, we plan to add it at the end of the year. We don't define a specific date, we prefer to use a three-months period instead (Q1, Q2, Q3, Q4).

**Comment by LucasMatuszewski** [comment]
@Aurelsicoko - thx a lot!
If we decide to use Strapi in alpha:
1. Is there some easy/automated way to update to beta/final version ? Or do we have to do all manually?
2. If we would made our own features, bugfixes or integrations, are you open for PR from everyone? :)

**Comment by Aurelsicoko** [comment]
1. You have to do it manually from alpha to beta. Then, the beta has a completely different files structure which will offer an easy way to upgrade from a version to another.

**Comment by LucasMatuszewski** [comment]
~2 weeks! Great to hear it. So we will start development with new version  :)

**Comment by PhantomxJoker** [comment]
Hi! Is there any update on this? Can I use this feature on alpha ?

**Comment by derrickmehaffy** [comment]
@PhantomxJoker no update yet but when it is released there won't be any alpha support only beta.

**Comment by Aurelsicoko** [comment]
Oh, I totally forgot this one! I added it to the list of pain points encountered as a user. Recently, we split the administrators and end users into two separate categories of users. It's typically the kind of issues we want to solve (see more https://blog.strapi.io/why-we-split-the-management-of-the-admin-users-and-end-users/).

**Comment by tomeady** [comment]
upvoteeee

**Comment by derrickmehaffy** [comment]
@tomeady there is no timeline on building this feature but more than likely not going to happen before November of this year as the Strapi team have already laid out their plans until then in their most recent blog post.

**Comment by botzill** [comment]
Hi. Any updates on this feature? 

I think feature is really important and will open a lot of other scenarios.

Thx a lot.

**Comment by derrickmehaffy** [comment]
@botzill No updates or ETA yet.

**Comment by JamesBrightman** [comment]
Q2 2020 looks like the latest time estimate according to https://portal.productboard.com/strapi/1-public-roadmap/tabs/3-planned.

Is the only way for users to enter/update records for them to log in via the admin panel? This seems like a huge oversight and makes Strapi extremely impractical for many uses.

**Comment by derrickmehaffy** [comment]
@JamesBrightman no they could do so with the proper permissions either via REST or GraphQL. 

Here is the blog post related to why this was done: https://strapi.io/blog/why-we-split-the-management-of-the-admin-users-and-end-users?redirectPage=1

**Comment by JamesBrightman** [comment]
@derrickmehaffy 

Thanks for the link, so if I have this right -

> An End User can only consume the API. This may include adding content but through the GET/POST methods in a form

So it is possible for a non admin to add (and edit?) content via API's. However I maintain that this is still a less than great solution given than the user may not be familiar with using APIs. I suppose i'll have to wait for a more solid UI solution.

**Comment by derrickmehaffy** [comment]
Right now the adminUI should not be used by normal users (at least not until that linked feature request is built). It's more so for the moment a simple way for admin level users to do light content-management.

Your best bet would be to handle content creation on your frontend or something tying into the generated APIs. The adminUI itself has it's own internal API that doesn't use the standard routes, controllers, services, lifecycle callbacks, or policies (again to allow for separation of admin level content additions vs user usable APIs). As far as I'm aware (and @soupette could confirm) there is no plan to change this either, though there is progress being made to allow users an easier time of modifying the admin and it's plugins as a whole.

**Comment by JamesBrightman** [comment]
@derrickmehaffy 
Ok, I think I understand your point. Users cannot view adminUI so instead we serve something like a simple web app which can allow users to enter content, which is then used behind the scenes via the strapi API to add records to the strapi content?

**Comment by derrickmehaffy** [comment]
Yes exactly

**Comment by JamesBrightman** [comment]
@derrickmehaffy 
That is definitely an Interesting solution. Do you happen to have an example of an update/create record API call?

**Comment by derrickmehaffy** [comment]
If you install the documentation plugin: https://www.npmjs.com/package/strapi-plugin-documentation it generates automated swagger docs. Likewise the graphql plugin has the built in graphql playground (which also has some built in docs based on your schema file)

**Comment by JamesBrightman** [comment]
I'll give it a shot, thanks.

**Comment by Chris533** [comment]
[react-admin](https://github.com/nazirov91/ra-strapi-rest) or [bunadmin](https://github.com/bunred/bunadmin-plugin-buncms-strapi-user) can make it for now.

**Comment by surgeharb** [comment]
![Screen Shot 2020-07-08 at 5 28 25 PM](https://user-images.githubusercontent.com/22060043/86931173-739da380-c140-11ea-95e6-f815d36378c4.png)

Would something like this make your life easier guys?
Working on Strapi v3, it will restrict access to non-admin users to specific entries in the left menu.

P.S. Also made some changes in the admin/src to hide "Manage Administrators" and remove "Unused Plugins" from routing.

**Comment by botzill** [comment]
@surgeharb It's really nice to have this feature. Let us know when you have more updates to test it.

**Comment by carlosmarinho** [comment]
That would be a great feature!
________________________________
De: Chirica Gheorghe <notifications@github.com>
Enviado: quarta-feira, 8 de julho de 2020 15:00
Para: strapi/strapi <strapi@noreply.github.com>
Cc: carlosmarinho <carluizfla@hotmail.com>; Mention <mention@noreply.github.com>
Assunto: Re: [strapi/strapi] Allow other User roles access to Admin Panel (#1152)


@surgeharb<https://github.com/surgeharb> It's really nice to have this feature. Let us know when you have more updates to test it.

**Comment by surgeharb** [comment]
@botzill, I will share it in the next 24hours on how I achieved this in an article or a GitHub repo.
Keep in mind that this will be a temporary solution as the core team of Strapi is in the latest stages of achieving this the good way in the system.
I made use of the built in Strapi "admin customization" feature.

**Comment by surgeharb** [comment]
@botzill @carlosmarinho @johackim @JamesBrightman @tomeady @PhantomxJoker 
As promised, this is how to achieve access rights in Strapi v3.
This is a temporary workaround that allows you to manage multiple roles for Strapi administrators.
Let me know what you think of this: [https://github.com/surgeharb/strapi-plugins#plugin-admin-access-rights](https://github.com/surgeharb/strapi-plugins#plugin-admin-access-rights)

**Comment by derrickmehaffy** [comment]
I'd like to point out to anyone watching this thread that the Strapi RBAC feature is currently in testing (PR is already open for it: https://github.com/strapi/strapi/pull/6965 ) and should be released shortly, however there is still a lot of things to review and test as I'm sure the entire Strapi team want to squash any bugs before users jump on migrating.

**Comment by surgeharb** [comment]
Amazing news @derrickmehaffy!
If I can assist in testing or fixing any bugs, I am more than happy to help.

**Comment by sebacampos** [comment]
> I'd like to point out to anyone watching this thread that the Strapi RBAC feature is currently in testing (PR is already open for it: #6965 ) and should be released shortly, however there is still a lot of things to review and test as I'm sure the entire Strapi team want to squash any bugs before users jump on migrating.

Hi! Will that RBAC feature be available only for EE version or also CE ? 
![image](https://user-images.githubusercontent.com/16866034/87863324-f89a7100-c927-11ea-865e-8d85afcffe69.png)

**Comment by surgeharb** [comment]
@sebacampos this is mentioned here https://portal.productboard.com/strapi/1-public-roadmap/c/8-users-roles-permissions-ce-ee, only 3 pre-defined roles are available by default for the Community Edition (CE), upgrading to the Enterprise Edition (EE) will allow customers to create an unlimited number of customizable roles.

<img width="739" alt="Screen Shot 2020-07-19 at 10 03 37 AM" src="https://user-images.githubusercontent.com/22060043/87869278-21158000-c9a7-11ea-8d13-6133ef067926.png">

**Comment by MonikaPatelIT** [comment]
@surgeharb that's perfect solution <3
